modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/devicemgtservice/DeviceManagementService.java

```diff
@@ -0,0 +1,158 @@
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.license.mgt.License;
+import org.wso2.carbon.mdm.mobileservices.windows.common.exceptions.WindowsConfigurationException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.Message;
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.WindowsAPIUtils;
+
+import javax.jws.WebService;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Windows Device Management REST-API implementation.
+ * All end points supports JSON, XMl with content negotiation.
+ */
+@WebService
+@Produces({"application/json", "application/xml"})
+@Consumes({"application/json", "application/xml"})
+public class DeviceManagementService {
+
+    private static Log log = LogFactory.getLog(DeviceManagementService.class);
+
+    /**
+     * Get all devices.Returns list of Windows devices registered in MDM.
+     *
+     * @return Device List
+     * @throws WindowsConfigurationException
+     */
+    @GET
+    public List<Device> getAllDevices()
+            throws WindowsConfigurationException {
+        String msg;
+        List<org.wso2.carbon.device.mgt.common.Device> devices;
+
+        try {
+            devices = WindowsAPIUtils.getDeviceManagementService().
+                    getAllDevices(DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_WINDOWS);
+        } catch (DeviceManagementException e) {
+            msg = "Error occurred while fetching the device list.";
+            log.error(msg, e);
+            throw new WindowsConfigurationException(msg, e);
+        }
+        return devices;
+    }
+
+    /**
+     * Fetch Windows device details of a given device Id.
+     *
+     * @param id Device Id
+     * @return Device
+     * @throws WindowsConfigurationException
+     */
+    @GET
+    @Path("{id}")
+    public org.wso2.carbon.device.mgt.common.Device getDevice(@PathParam("id") String id)
+            throws WindowsConfigurationException {
+
+        String msg;
+        org.wso2.carbon.device.mgt.common.Device device;
+
+        try {
+            DeviceIdentifier deviceIdentifier = WindowsAPIUtils.convertToDeviceIdentifierObject(id);
+            device = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
+            if (device == null) {
+                Response.status(Response.Status.NOT_FOUND);
+            }
+        } catch (DeviceManagementException deviceMgtEx) {
+            msg = "Error occurred while fetching the device information.";
+            log.error(msg, deviceMgtEx);
+            throw new WindowsConfigurationException(msg, deviceMgtEx);
+        }
+        return device;
+    }
+
+    /**
+     * Update Windows device details of given device id.
+     *
+     * @param id     Device Id
+     * @param device Device Details
+     * @return Message
+     * @throws WindowsConfigurationException
```

Comment:
As per Harshan's comments, It is better if the return values and exception (reason for exception etc) are described well.